configure.ac

```diff
@@ -333,31 +333,36 @@ if test x$use_sanitizers != x; then
     ]],[[]])])
 fi
 
+dnl APPEND_FLAG_IF_AVAILABLE([APPEND_TO_THIS_VARIABLE], [-WallExtra])
+AC_DEFUN([APPEND_FLAG_IF_AVAILABLE], [
```

Comment:
Could also define `SET_FLAG_IF_AVAILABLE` and use throughout.

Reply:
That would be a little bit less straight forward. While such code would definitely benefit from it:
```
AX_CHECK_COMPILE_FLAG([-msse4.2],[[SSE42_CXXFLAGS="-msse4.2"]],,[[$CXXFLAG_WERROR]])
```
and become:
```
SET_FLAG_IF_AVAILABLE([SSE42_CXXFLAGS], [-msse4.2])
```
there are other usages where the code calls `AC_MSG_ERROR()` if the flag is not available or does not compile with an additional `$CXXFLAG_WERROR`.

I will consider this as a separate PR.

Reply:
Yeah didn't check if all follow the same pattern. Different PR is fine.